src/Worker/CentrifugeWorker.php

```diff
@@ -0,0 +1,78 @@
+use Baldinof\RoadRunnerBundle\Event\Centrifuge\RefreshEvent;
+use Baldinof\RoadRunnerBundle\Event\Centrifuge\RPCEvent;
+use Baldinof\RoadRunnerBundle\Event\Centrifuge\SubRefreshEvent;
+use Baldinof\RoadRunnerBundle\Event\Centrifuge\SubscribeEvent;
+use Psr\Log\LoggerInterface;
+use RoadRunner\Centrifugo\CentrifugoWorker;
+use RoadRunner\Centrifugo\Payload\ConnectResponse;
+use RoadRunner\Centrifugo\Payload\PublishResponse;
+use RoadRunner\Centrifugo\Payload\RefreshResponse;
+use RoadRunner\Centrifugo\Payload\RPCResponse;
+use RoadRunner\Centrifugo\Payload\SubRefreshResponse;
+use RoadRunner\Centrifugo\Payload\SubscribeResponse;
+use RoadRunner\Centrifugo\Request;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class CentrifugeWorker implements WorkerInterface
+{
+    public function __construct(
+        private LoggerInterface $logger,
+        private CentrifugoWorker $centrifugoWorker,
+        private EventDispatcherInterface $eventDispatcher,
+        private HttpDependencies $httpDependencies,
+    ) {
+    }
+
+    public function start(): void
+    {
+        $this->logger->debug('Centrifuge worker started');
+
+        while ($request = $this->centrifugoWorker->waitRequest()) {
+            $eventClass = match (true) {
+                $request instanceof Request\Connect => ConnectEvent::class,
+                $request instanceof Request\Publish => PublishEvent::class,
+                $request instanceof Request\Refresh => RefreshEvent::class,
+                $request instanceof Request\SubRefresh => SubRefreshEvent::class,
+                $request instanceof Request\Subscribe => SubscribeEvent::class,
+                $request instanceof Request\RPC => RPCEvent::class,
+                $request instanceof Request\Invalid => InvalidEvent::class,
+                default => throw new \RuntimeException(sprintf("Unsupported \$request type '%s'", $request::class)),
+            };
+
+            $event = new $eventClass($request);
+
+            try {
+                $processedEvent = $this->eventDispatcher->dispatch($event);
+                \assert($processedEvent instanceof CentrifugeEventInterface);
+
+                $response = $processedEvent->getResponse() ?? match ($eventClass) {
+                    ConnectEvent::class => new ConnectResponse(),
+                    PublishEvent::class => new PublishResponse(),
+                    RefreshEvent::class => new RefreshResponse(),
+                    SubRefreshEvent::class => new SubRefreshResponse(),
+                    SubscribeEvent::class => new SubscribeResponse(),
+                    RPCEvent::class => new RPCResponse(),
+                    default => null,
+                };
+
+                if ($response !== null) {
+                    $request->respond($response);
+                }
+            } catch (\Throwable $throwable) {
+                $request->disconnect(500, 'Server error');
```

Comment:
Should we do more stuff here?

In the http context, exception are converted into http responses (unles something happen of the Symfony request handling), so we almost never reach this catch in the http worker.

Maybe we should re-use the `kernel_reboot.allowed_exceptions` list and do not disconnect if it's whitelisted?

Reply:
In the context of websocket I think we should always disconnect on any exception by default. You are expected to send valid payload that the other side can parse without errors, since there is no "default" payload that you can send that will be always processed. It's up to the user to handle his own errors as they can always change the response event to something else. Centrifuge has also auto reconnect feature that handles disconnects for you.